source/grid/grid_tools_topology.cc

```diff
@@ -1855,7 +1855,10 @@ namespace GridTools
     const Triangulation<dim, spacedim> &triangulation,
     DynamicSparsityPattern             &cell_connectivity)
   {
-    // Assume at most 16 neighbors per cell (a structured hex mesh has 8)
+    // In a structured hex mesh, about 98% of vertices have 16 neighbors or
+    // fewer. Similarly, in an unstructured tet mesh, about 86% of vertices have
+    // 16 neighbors or fewer. Hence, pick 16 neighbors here to cover most cases
+    // without allocation.
```

Comment:
I know that in a year, I will wonder whether there are theoretical arguments for these numbers and will try to derive your 98% and 86% numbers on a piece of paper. But these are just *empirical* numbers. Would you mention this somewhere to make that clear?

In fact, I think it would be great to actually copy-paste the numbers you showed in the other PR as the basis for the numbers you mention? I bet that at some point in the future someone will use these numbers for something else, without having to re-do your experiment.

Reply:
Good idea - we have the data, after all.